bee-message/src/output/signature_locked_asset.rs

```diff
@@ -0,0 +1,182 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{address::Address, payload::PAYLOAD_LENGTH_MAX, MessagePackError, MessageUnpackError, ValidationError};
+
+use bee_packable::{
+    error::{PackPrefixError, UnpackPrefixError},
+    PackError, Packable, Packer, UnknownTagError, UnpackError, Unpacker, VecPrefix,
+};
+
+use alloc::vec::Vec;
+use core::{
+    convert::{Infallible, TryInto},
+    fmt,
+};
+
+const ASSET_ID_LENGTH: usize = 32;
```

Comment:
consider creating an `AssetId` type, which would make the API nicer and allows us to impl `Debug`/`Display` and maybe other niceties on it.